internal/pkg/agent/cmd/run.go

```diff
@@ -109,11 +109,15 @@ func run(override cfgOverrider, modifiers ...component.PlatformModifier) error {
 	if cfg.Settings.LoggingConfig != nil {
 		logLvl = cfg.Settings.LoggingConfig.Level
 	}
-	l, err := logger.NewFromConfig("", cfg.Settings.LoggingConfig, true)
+	baseLogger, err := logger.NewFromConfig("", cfg.Settings.LoggingConfig, true)
 	if err != nil {
 		return err
 	}
 
+	l := baseLogger.With("component", map[string]interface{}{
+		"binary": agentName,
```

Comment:
As a bonus the mappings for this on the Fleet side are now already in place: https://github.com/elastic/integrations/pull/4888/files